Allow using Promise.finally

The repo uses ES2015, which makes using `Promise.finally` forbidden.

As this got introduced in ES2018, this would be nice if the lib can be updated to use ES2018.

Currently:

```shell
$ make build
if [ ! -d node_modules ]; then npm install; fi;
tsc
src/index.ts:19:6 - error TS2339: Property 'finally' does not exist on type 'Promise<void>'.

19     .finally ( () => {});
        ~~~~~~~

make: *** [Makefile:11: build] Error 2
```

For context, I stumbled into this for https://github.com/jenkins-infra/evergreen/pull/300 where I am using Promise.finally, and thought I'd update it here too.

--- a/src/index.ts
+++ b/src/index.ts
@@ -12,3 +12,9 @@ process.on('unhandledRejection', (reason, p) =>
 server.on('listening', () =>
   logger.info('Feathers application started on http://%s:%d', app.get('host'), port)
 );
+
+// REMOVE ME: this part just shows/checks that we can use Promise.finally, introduced in ES2018.
+Promise.resolve('blah')
+    .then( () => {})
+    .finally ( () => {});
+// END REMOVE ME

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -4,7 +4,7 @@
     "outDir": "./build",
     "module" : "commonjs",
     "skipLibCheck": true,
-    "lib" : ["es2017"],
+    "lib" : ["es2018"],
     "module": "commonjs",
     "moduleResolution": "node",
     "esModuleInterop": true,

